contracts/paramstore/src/contract.rs

```diff
@@ -0,0 +1,74 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult, Storage,
+};
+use cw2::set_contract_version;
+
+use crate::error::ContractError;
+use crate::msg::{InstantiateMsg, ParamResponseu32, QueryMsg, SudoMsg};
+use crate::state::U32;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:paramstore";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+// TODO: make sure this can only be sudo initialized
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute("owner", info.sender))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn sudo(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: SudoMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        SudoMsg::UpdateParamu32 {
+            contract_name,
+            key,
+            value,
+        } => {
+            let param = U32.update(
+                deps.storage,
+                (contract_name.clone(), key.clone()),
+                |old| -> StdResult<_> { Ok(old.unwrap_or_default() + value) },
```

Comment:
is this doing an increment? I think it should only set the value right?

Reply:
This should set the value if it doesn't exist or update it if it does. It currently doesn't do the latter. Have to write tests :)